tests/test_operator.py

```diff
@@ -2011,10 +2011,10 @@ def test_indirection(self):
         op = Operator(eqns)
 
         assert op._dspace[time].lower == 0
-        assert op._dspace[time].upper == 1
-        assert op.arguments()['time_M'] == nt - 2
+        assert op._dspace[time].upper == 0
+        assert op.arguments()['time_M'] == nt - 1
```

Comment:
at first glance, this looks wrong?

if `f` has `nt` slots, then it can be written up to `nt-2` included no? at `nt-2`, `f` will be written at `f[s, ...] = f[time+1, ...] = f[nt-2+1, ...] = f[nt-1, ...]` which is the extreme

So this looks borked